### PR TITLE
ISSUE #2258: Update Authorizing Official to Senior Official

### LINKED ISSUE #2258: `Senior Official` replaces `Authorizing Official` in registrar
### Issue description

Because the term `Authorizing Official` or AO has special meaning in the Federal government, we must refer to the role in the registrar and public-facing website differently.

As a result of #2096, we have determined that the term `Senior Official`, with the abbreviation SO, will replace the use of `Authorizing Official` everywhere in the code and user-facing content.

### Acceptance criteria

- [ ] All instances of `Authorizing Official` are replaced with `Senior Official` in code and content.

### Additional context

Rather than Product finding all instances of this content, it is recommended that dev performs the appropriate search through the entire code-base, to identify each occurrence, replacing as appropriate. Also, any context around that change must be reviewed, so that it continues to make sense to all parties.

Please make sure to have a normal design review and make any content changes. This review will likely take longer than typical.

### Links to other issues

Based on:
- [x] #2096 
Related to:
- [ ] #2258 (This ticket)
- [ ] #2259

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #2258 

## Changes

* Rename `Authorizing Official` to `Senior Official` everywhere, including unit tests and in file names (you'll see it renamed from `ao` to `so`)

## Setup

1. Go onto [my sandbox ](getgov-rh.app.cloud.gov)
2. Make sure you see `senior_official` in the URL routes, `Senior Official` in the request texts and in `/admin` and more
Screenshot 2024-06-26 at 1 49 51 PM
Screenshot 2024-06-26 at 1 55 59 PM

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/templates/domain_senior_official.html [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
   {# this is right after the messages block in the parent template #}
   {% include "includes/form_errors.html" with form=form %}
 
-  <h1>Authorizing official</h1>
+  <h1>Senior official</h1>
 
-  <p>Your authorizing official is a person within your organization who can
-  authorize domain requests. This person must be in a role of significant, executive responsibility within the organization. Read more about <a class="usa-link" rel="noopener noreferrer" target="_blank" href="{% public_site_url 'domains/eligibility/#you-must-have-approval-from-an-authorizing-official-within-your-organization' %}">who can serve as an authorizing official</a>.</p>
+  <p>Your senior official is a person within your organization who can
+  authorize domain requests. This person must be in a role of significant, executive responsibility within the organization. Read more about <a class="usa-link" rel="noopener noreferrer" target="_blank" href="{% public_site_url 'domains/eligibility/#you-must-have-approval-from-a-senior-official-within-your-organization' %}">who can serve as a senior official</a>.</p>
```
[the remark this answers is not in the document]
Comment (therealslimhsiehdy) [reply src/registrar/templates/domain_senior_official.html:13]: That is 100% correct!

File: src/registrar/templates/emails/status_change_rejected.txt [hunk 2]
```diff
@@ -45,7 +45,7 @@ government organizations.
 
 DEMONSTRATE ELIGIBILITY
 If you can provide documentation that demonstrates your eligibility, reply to this email. 
-This can include links to (or copies of) your authorizing legislation, your founding 
+This can include links to (or copies of) your senior legislation, your founding
```
[the remark this answers is not in the document]
Comment (therealslimhsiehdy) [reply src/registrar/templates/emails/status_change_rejected.txt]: This SVG changes are in [this commit](b8d1b5cf9cbdb92a6bf515a2b291a553fa156b44) -- I chatted with Zander and I moved it back to `authorizing legislation` bc that seems more like the correct wording? Up to Katherine though as I'm not sure

[the remark this answers is not in the document]
Comment (Katherine-Osos) [reply src/registrar/templates/emails/status_change_rejected.txt]: Correct, that instance should not change. Should still read "authorizing legislation."

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

**Pull request merged**